docs/spec/spec-proposals/f1-fee-distribution/f1_fee_distr.tex

```diff
@@ -85,36 +105,61 @@ \subsection{Slashing}
+
+Make each block store the evaluation of $inflation$, which represents the number of staking tokens a single staked token should produce. 
+At the end of each period, increase the validators total number of tokens within the fee distribution by the amount due to inflation. This can be done by multiplying the validators number of staked tokens by $(1 + $ sum of x over the period$)$.
 
-Make each block have an inflation number, by which every staked token should produce $x$ additional staking tokens. In state there is a variable for the sum of all such inflation numbers. Then each period will store this total inflation sum in addition to $\sum_{i=0}^{end}\frac{T_i}{n_i}$. When withdrawing perform a subtraction on the inflation sums at the end and at the start to see how many new staking tokens to produce per staked token. 
+In state a variable should be kept for the sum of all $inflation$ evaluations from now until genesis.
+Then each period will store this total inflation sum in addition to what it already stores per-period.
+The presence of comission rates causes the distribution of delegated stake within a validator to change per block.
+To get around this, when accumulating the value per validator in state, instead of calculating the rewards you'd get for 1 stake, you treat it as though you have $\prod_0^{now} 1 + x$ stake, where $\prod_0^{now} 1 + x$ is the amount of stake accumulated since genesis until now if you had $1$ stake originally bonded.
+When withdrawing, you take the difference as before, which yields the amount of rewards you would have obtained with $(\prod_0^{begin\ bonding\ period}1 + x)$ stake from the block you began bonding at until now. You then divide that by $(\prod_0^{begin\ bonding\ period}1 + x)$ to normalize it, and then proceed as before.
 
-This works great in the model where the inflation rate should be dynamic each block, but apply the same to each validator. Inflation creation can trivially be epoched as long as inflation isn't required within the epoch, through changes to the $x$ function.
+TODO: Explain this better / put equations to prove correctness
+
+This works great in the model where the inflation rate should be dynamic each block, but apply the same to each validator.
+Inflation creation can trivially be epoched as long as inflation isn't required within the epoch, through changes to the $inflation$ function.
 
 Note that this process is extremely efficient.
 
-The above can be trivially amended if we want inflation to proceed differently for different validators each block. (e.g. depending on who was offline) It can also be made to be easily adapted in a live chain. It is unclear if either of these two are more desirable settings.
+The above can be trivially amended if we want inflation to proceed differently for different validators each block.
```

Comment:
What is the trivial amendment? This seems like it would make tracking `pi(1 + x)` from `0` to `now` more complex.

Reply:
It would be tracking it per validator, instead of one global entry. Its a linear overhead in state, which can be a struct in the same address which offsets the costly part of state.

Reply:
OK, makes sense. I still think varying the rate instead of the amount doesn't have too much utility, but it's never harmful to have the option.